smtp/connection.ts

```diff
@@ -386,15 +386,27 @@ export class SMTPConnection extends EventEmitter {
 			? [codes]
 			: [250];
 
+		let hasResentForGreylist = false;
```

Comment:
is this problematic if you are sending multiple emails in a sequence of send calls, and then 450 is returned on all of them? be nice if this was tracked internally to the `once` pattern we desire here

Reply:
from what i understood `response` is created per-request, so this is isolated to a given message. i agree it would be cleaner to use an event-driven pattern instead. not sure how to refactor for that, though.

Reply:
with the closure variable, `RCPT TO` was being called three times; switching to a WeakMap approach reduced that to just once. good call!